### PR TITLE
add support for cron mode and quiet mode. clean up and organize usage.

### PR DESCRIPTION
**Pull request by ajmiller82**
Added support for a quiet mode for anyone who wants to integrate with a wrapper script and only check exit codes. Added support for a cron mode that only creates messages to stdout on actionable information for cron email notification on updates. Cleaned up and organized app usage.

### CODE REVIEW

**Comment by devinslick** [comment]
Great idea!  We might also need to add some information about this in the README.

File: plexupdate.sh [hunk 1]
```diff
@@ -85,50 +87,77 @@ URL_LOGIN=https://plex.tv/users/sign_in
...
+                (--) ;;
+                (-*) echo "Error: unrecognized option $1" 1>&2; exit 1;;
+                (*)  break;;
 	esac
 	shift
 done
 
+# send all stdout to /dev/null
+if [ "${QUIET}" = "yes" ]; then
+        exec 1> /dev/null
```
Comment (mrworf) [inline plexupdate.sh]: Should stderr also be redirected? According to the fix, QUIET turns off all output and only returns errorcodes while CRON essentially disables STDOUT.

If so, you could do 1> for CRON and 1> 2> for QUIET, no? Would avoid all extra CRON if/then statements too.

File: plexupdate.sh [hunk 2]
```diff
@@ -85,50 +87,77 @@ URL_LOGIN=https://plex.tv/users/sign_in
...
+        echo "    -a Auto install if download was successful (requires root)"
+        echo "    -C Cron mode. Only output to stdout on an actionable operation"
+        echo "    -d Auto delete after auto install"
+        echo "    -f Force download even if it's the same version or file"
+        echo "       already exists (WILL NOT OVERWRITE)"
+        echo "    -h This help"
+        echo "    -k Reuse last authentication"
+        echo "    -o 32-bit version (default 64 bit)"
+        echo "    -p Public Plex Media Server version"
+        echo "    -q Quiet mode. No stdout/stderr, only exit codes"
```
Comment (mrworf) [inline plexupdate.sh]: This is not true since quiet won't suppress errors in the current patch ?

**Comment by mrworf** [comment]
Review the comment and let me know.

File: plexupdate.sh (hunk 1 above)
Comment (ajmiller82) [inline plexupdate.sh]: The way I see it, anything stderr should always be seen, even with quiet mode on. The intent isn't to suppress everything but exit codes, but to suppress informational-only output on successful runs.

I agree the if statements aren't very elegant, but the alternative would be writing a logger function to handle the different information level states.

File: plexupdate.sh (hunk 2 above)
Comment (ajmiller82) [inline plexupdate.sh]: Ah. No wonder I confused you. My intent was never to include stderr.

**Comment by mrworf** [comment]
Fix the arguments and rebase and I'll add it :)

**Comment by ajmiller82** [comment]
Fixed help text. Updated README.md. Added a silent mode argument (because why not?).

**Pull request merged**